lib/phase/flatten.js

```diff
@@ -0,0 +1,88 @@
+'use strict';
+
+const Hoek = require('hoek');
+const clone = require('clone');
+
+/**
+ * Overlays the stuff specified in a specific job on top of the defaults in shared
+ *
+ * @method flattenSharedIntoJobs
+ * @param  {Job}          shared A kind of default Job template
+ * @param  {Object}       jobs   List of jobs (name => job)
+ * @return {Object}              Updated list of jobs after merging
+ */
+function flattenSharedIntoJobs(shared, jobs) {
+    const newJobs = {};
+
+    Object.keys(jobs).forEach((jobName) => {
+        const newJob = clone(shared);
+        const oldJob = clone(jobs[jobName]);
+
+        ['image', 'matrix', 'steps'].forEach((key) => {
+            if (oldJob[key]) {
+                newJob[key] = oldJob[key];
+            }
+        });
+
+        if (!newJob.environment) {
+            newJob.environment = {};
```

Comment:
Ahh, it's because in the next two lines we merge the job's environment on top of shared.  Shared needs to be overwritten.